waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -86,12 +97,26 @@ constructAppSpec waspDir options decls = do
     [] -> Right appSpec
     validationErrors -> Left $ map show validationErrors
 
-findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Either String (Path' Abs File'))
+findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
 findWaspFile waspDir = do
   files <- fst <$> IOUtil.listDirectory waspDir
-  return $ maybeToEither "Couldn't find a single *.wasp file." $ (waspDir </>) <$> find isWaspFile files
+  return $ (waspDir </>) <$> find isWaspFile files
   where
     isWaspFile path =
       ".wasp"
         `isSuffixOf` toFilePath path
         && (length (toFilePath path) > length (".wasp" :: String))
+
+analyzePackageJsonContent :: Path' Abs (Dir WaspProjectDir) -> IO (Either [CompileError] PackageJson)
+analyzePackageJsonContent waspProjectDir =
+  findPackageJsonFile >>= \case
+    Just packageJsonFile -> readPackageJsonFile packageJsonFile
+    Nothing -> return $ Left [fileNotFoundMessage]
+  where
+    fileNotFoundMessage = "couldn't find package.json file in the " ++ toFilePath waspProjectDir ++ " directory"
+    findPackageJsonFile = findFileInWaspProjectDir waspProjectDir [relfile|package.json|]
+
+readPackageJsonFile :: Path' Abs File' -> IO (Either [CompileError] PackageJson)
+readPackageJsonFile packageJsonFile = do
+  byteString <- IOUtil.readFileBytes packageJsonFile
+  return $ maybeToEither ["Error reading the package.json file"] $ Aeson.decode byteString
```

Comment:
```suggestion
  return $ maybeToEither ["Error decoding the package.json file"] $ Aeson.decode byteString
```

Reply:
Hm, not sure about this one. "Decoding" sounds very technical, while "Reading" implies that the file wasn't found. Perhaps "parsing" is the best choice?

Reply:
I agree!

Reply:
Verdict: we'll change it to `Reading`.

Reply:
I changed it to `parsing`.